### PR TITLE
Implement Analytics

### LINKED ISSUE #219: Implement new analytics
This ticket is to 

- implement the missing analytics based on #216 
- rename the existing analytics to make them easier to understand

See the `To do` column of this [page on confluence](https://confluence.dev.bbc.co.uk/display/newslabs/Analytics+overview) to see what needs to done for each analytic.

**Example**
For creating a new project, the  existing code (Projects/index.js line 58) should be renamed to this:
```    
props.trackEvent({ category: 'Projects', action: `Create project with id: ${ item.id }` });
```

**Acceptance criteria**

- [ ] Implementation of missing analytics and renaming of existing ones complete (outlined on [analytics confluence page](https://confluence.dev.bbc.co.uk/display/newslabs/Analytics+overview))
- [ ] Analytics confluence page `To do` column updated/removed if nothing more to be done

To check the changes, you can do so either from within the network tab:
Screenshot 2021-05-25 at 11 52 42

Or via the [matomo console](https://newslabs-analytics.tools.bbc.co.uk/index.php?module=CoreHome&action=index&date=last7&period=range&idSite=2#?idSite=2&period=range&date=last7&segment=&category=Dashboard_Dashboard&subcategory=1)

### PR DESCRIPTION
**Pull request by sarahrainbow**
Fixes #219  

Linked to this [storybook PR ](https://github.com/bbc/digital-paper-edit-storybook/pull/109)

This PR is to

- implement the missing analytics based on #216
- rename the existing analytics to make them easier to understand
- fixes annoying compilation warnings 

See this page on [confluence](https://confluence.dev.bbc.co.uk/display/newslabs/Analytics+overview) to see what analytics should now be in place.

## Acceptance Criteria
- Analytics are implemented and appropriately named
- `npm start` should no longer give any compilation warnings 

## review tips
- checkout 219-analytics in firebase **and** storybook
- npm link storybook and firebase
- To check the analytics, you can do so from within the network tab:
Screenshot 2021-05-25 at 11 52 42

### CODE REVIEW

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/dist/programme-script-json-to-docx.dev.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+'use strict';
```
Comment (tamsingreen) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/dist/programme-script-json-to-docx.dev.js:1]: I think this transpiled version might have been included by mistake

File: src/App.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ const App = (props) => {
   const [ user, setUser ] = useState();
 
   useEffect(() => {
+    // eslint-disable-next-line no-shadow
     const authListener = props.firebase.auth.onAuthStateChanged((user) =>
```
Comment (tamsingreen) [inline src/App.js:21]: Can just remove the user param here to fix the lint error

File: src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/index.js [hunk 3]
```diff
@@ -18,13 +18,7 @@ import { formatJsonToText, getJson } from './json';
...
 import programmeScriptJsonToDocx from './programme-script-json-to-docx';
 
-const ExportDropdown = (props) => {
-
-  const title = props.title;
-  const elements = props.elements;
-  const transcripts = props.transcripts;
-  const projectTitle = props.projectTitle;
-  const storage = props.storage;
+const ExportDropdown = ({ title, elements, transcripts, projectTitle, storage, trackEvent, handleGetMediaUrl }) => {
```
Comment (tamsingreen) [inline src/Components/PaperEditor/ProgrammeScriptContainer/ExportDropdown/index.js:21]: 👌🏻

File: src/HelpOverlayTrigger.js [hunk 4]
```diff
@@ -15,21 +17,27 @@ const HelpOverlayTrigger = () => {
...
+          onClick={ () => trackEvent({ category: 'global - help dropdown', action: 'click', name: 'view user manual' }) }
         >
           User manual.
         </a>
         <br />
         This is a <strong>slightly different version</strong> of the Digital
         Paper Edit, but the general workflows should be the same.
         Should you require login details, or more assistance,{' '}
-        <a href="mailto:dpe@bbcnewslabs.co.uk">get in touch with Newslabs</a>!
+        <a
```
Comment (tamsingreen) [inline src/HelpOverlayTrigger.js:28]: Alli's updated this code recently with the new user manual so could you pull in her changes from master?

**Review by tamsingreen — CHANGES_REQUESTED** [review]
Looks good - couple of tweaks to make before merging bar any further feedback from the meeting :)

File: src/App.js (hunk 2 above)
Comment (tamsingreen) [reply src/App.js:21]: This is NOT the right answer, ignore me @sarahrainbow

File: src/Components/PaperEditor/ProgrammeScriptContainer/index.js [hunk 5]
```diff
@@ -55,11 +55,21 @@ const ProgrammeScriptContainer = (props) => {
...
+
+        return element.index === insertPointElement.index - 1;
+      });
       console.log('Successfully saved');
+
+      let insertedWords = '';
+      insertedPaperCut.words.forEach(word => {
+        insertedWords += word.text;
+      });
+      props.trackEvent({ category: 'programme script - programme script panel', action: 'insert selection', name: insertedWords });
```
Comment (allishultes) [inline src/Components/PaperEditor/ProgrammeScriptContainer/index.js:72]: Very detailed tracking 👍

File: src/Components/Workspace/PaperEdits/index.js [hunk 6]
```diff
@@ -48,7 +49,8 @@ PaperEdits.propTypes = {
   handleDuplicateItem: PropTypes.any,
   handleEditItem: PropTypes.any,
   items: PropTypes.any,
-  projectId: PropTypes.any
+  projectId: PropTypes.any,
+  trackEvent: PropTypes.any
```
Comment (allishultes) [inline src/Components/Workspace/PaperEdits/index.js:53]: Should this be 

```suggestion
  trackEvent: PropTypes.func
```
?

**Review by allishultes — APPROVED** [review]
Looks reallyy good, pal! Couple itty bitty comments. I haven't been super thorough in scrutinising all of the network calls locally but lmk if that would be helpful!

File: src/Components/Workspace/index.js [hunk 7]
```diff
@@ -415,12 +416,19 @@ const WorkspaceView = props => {
     return () => {};
   }, [ loadingPE, paperEditItems, id, props.firebase ]);
 
+  const convertMediaButton = () => {
```
[the remark this answers is not in the document]
Comment (sarahrainbow) [reply src/Components/Workspace/index.js:419]: nice suggestion – will rename 👍

File: src/Components/Workspace/PaperEdits/index.js (hunk 6 above)
Comment (sarahrainbow) [reply src/Components/Workspace/PaperEdits/index.js:53]: good spot!

**Pull request merged**